projects/storefrontapp/src/environments/models/feature.environment.model.ts

```diff
@@ -0,0 +1,6 @@
+import { ModuleWithProviders } from '@angular/core';
+
+export interface FeatureEnvironment {
+  enabled: boolean;
+  feature?: ModuleWithProviders;
```

Comment:
What when the module to be loaded is not necesarily a result of a static method (like forRoot()), but only a simple module class, will this typing work?

Reply:
Good point, we can use `Type<any> | ModuleWithProviders<any>` in that case.